lib/pprof/pprof.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pprof
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+)
+
+type PprofPort int
+
+const basePort = 6060
+
+const (
+	VCHInitPort PprofPort = iota
+	ImagecPort
+	VicadminPort
+	DockerPort
+	PortlayerPort
+	maxPort
+)
+
+var (
+	vchConfig metadata.VirtualContainerHostConfigSpec
+)
+
+func init() {
+	// load the vch config
+	// TODO: Optimize this to just pull the fields we need...
+	src, err := extraconfig.GuestInfoSource()
+	if err != nil {
+		log.Errorf("Unable to load configuration from guestinfo")
+		return
+	}
+	extraconfig.Decode(src, &vchConfig)
+}
+
+func StartPprof(name string, component PprofPort) error {
+	port := component + basePort
+	if component != ImagecPort {
+		log.Info(fmt.Sprintf("Launching %s server on port %d", name, port))
+	}
+
+	ip := "127.0.0.1"
+	self, ok := vchConfig.ExecutorConfig.Sessions["vicadmin"]
```

Comment:
Despite me having written this sample as reference, on a second read embedding "vicadmin" like this is very ugly. Let's drop the self portion and just check `vchConfig.ExecutorConfig.Diagnostics.DebugLevel`